### PR TITLE
hv: fix cpuid 0x2 mismatch when launch RTVM use atom core

### PR DESCRIPTION
**Pull request by tw4452852**
When CPUID executes with EAX set to 02H, the processor returns information about cache and TLB information.
This information is percpu related, and should be obtained directly from the physical cpu.

BTW, this patch is backported from v2.7 branch.

Tracked-On: #6931
Signed-off-by: Tw <wei.tan@intel.com>
Reviewed-by: Fei Li <fei1.li@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

File: hypervisor/arch/x86/guest/vcpuid.c [hunk 1]
```diff
@@ -850,11 +855,11 @@ void guest_cpuid(struct acrn_vcpu *vcpu, uint32_t *eax, uint32_t *ebx, uint32_t
...
 			/*
-			 * In this switch statement, leaf shall either be 0x01U or 0x0bU
-			 * or 0x0dU or 0x80000001U. All the other cases have been handled properly
-			 * before this switch statement.
-			 * Gracefully return if prior case clauses have not been met.
+			 * In this switch statement, leaf 0x01/0x0b/0x0d/0x19/0x80000001
+			 * shall be handled specifically. All the other cases
+			 * just return physical value.
 			 */
+			guest_cpuid_from_physical(leaf, eax, ebx, ecx, edx);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vcpuid.c:862]: This kind of emulation method is not the 1st time.  If we need to wrap in an API, we should do for all of them. But Given that is one line API. We can stay with direct call.

**Review by dongyaozu — CHANGES_REQUESTED** [review]

**Review by dongyaozu — APPROVED** [review]

File: hypervisor/arch/x86/guest/vcpuid.c [hunk 2]
```diff
@@ -785,6 +784,12 @@ static void guest_cpuid_80000001h(const struct acrn_vcpu *vcpu,
 	}
 }
 
+static void guest_cpuid_from_physical(uint32_t leaf,
```
Comment (lifeix) [inline hypervisor/arch/x86/guest/vcpuid.c:787]: static inline  or just call cpuid_subleaf in guest_limit_cpuid directly ?

**Comment by GGGZZZRRR** [comment]
@tw4452852 please check this build error
/home/acrn/workspace/acrn-hypervisor-build-native_9298/acrn-hypervisor/build/hypervisor/arch/x86/guest/vcpuid.o
arch/x86/guest/vcpuid.c: In function ‘guest_cpuid’:
arch/x86/guest/vcpuid.c:856:24: error: passing argument 2 of ‘cpuid_subleaf’ makes integer from pointer without a cast [-Werror=int-conversion]
  856 |    cpuid_subleaf(leaf, eax, ebx, ecx, edx);
      |                        ^~~
      |                        |
      |                        uint32_t * {aka unsigned int *}
In file included from arch/x86/guest/vcpuid.c:12:
/home/acrn/workspace/acrn-hypervisor-build-native_9298/acrn-hypervisor/hypervisor/include/arch/x86/asm/cpuid.h:156:58: note: expected ‘uint32_t’ {aka ‘unsigned int’} but argument is of type ‘uint32_t *’ {aka ‘unsigned int *’}
  156 | static inline void cpuid_subleaf(uint32_t leaf, uint32_t subleaf,
      |                                                 ~~~~~~~~~^~~~~~~
arch/x86/guest/vcpuid.c:856:4: error: too few arguments to function ‘cpuid_subleaf’
  856 |    cpuid_subleaf(leaf, eax, ebx, ecx, edx);
      |    ^~~~~~~~~~~~~
In file included from arch/x86/guest/vcpuid.c:12:
/home/acrn/workspace/acrn-hypervisor-build-native_9298/acrn-hypervisor/hypervisor/include/arch/x86/asm/cpuid.h:156:20: note: declared here
  156 | static inline void cpuid_subleaf(uint32_t leaf, uint32_t subleaf,
      |                    ^~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:548: /home/acrn/workspace/acrn-hypervisor-build-native_9298/acrn-hypervisor/build/hypervisor/arch/x86/guest/vcpuid.o] Error 1
make[1]: Leaving directory '/home/acrn/workspace/acrn-hypervisor-build-native_9298/acrn-hypervisor/hypervisor'
make: *** [Makefile:126: hypervisor] Error 2

**Review by lifeix — APPROVED** [review]

**Pull request merged**